### PR TITLE
Add support for new wQU225EC60to30 mesh

### PR DESCRIPTION
**Pull request by sbrus89**
- wQU225EC60to30 is a rotated grid which enables complete Arctic coverage without a pole singularity.
 - The wQU225EC60to30 vertices match the EC60to30v3 ocean/ice mesh cell centers in shallow regions, but the resolution grades out to 225km in the open ocean.
 - WW3 point output is now enabled via providing a stations file in the ww3_in namelist
 - Use WW3 routines to set output variable flags
 -  A new datm resolution has been added in order to compare with standalone WW3 runs with CFSR wind data (compset: `UWAV-CFSR`)
 - w2x variables can now be computed at the coupling interval independently from field output to file
 - Hooks for Stokes drift coupling to the ocean have been put into place.
 - A new compset `GWAV-IAF` has been added for a G case+WW3

### CODE REVIEW

**Comment by sbrus89** [comment]
Here are some hindcast results with the new wQU225EC60to30 mesh for June 2005 using 0.5 degree CFSR winds. Legend is as follows:

 - wQUEC225EC60to30_E3SM: Results from WW3 running with rotated pole in E3SM with datm and stubs for all other components
 - wQUEC225EC60to30_RTD: Results from standalone WW3 with rotated pole
 - QU100: Standalone WW3 results from a 100km global mesh with standard pole (no Arctic coverage)

These show that the wQUEC225EC60to30 mesh results are consistent between E3SM and standalone WW3 and that the rotated pole gives equivalent results to a standard pole mesh.

**Comment by sbrus89** [comment]
Here are a couple images of the wQU225EC60to30 mesh:

ocn_wav_mesh_bathytopo_globe
ocn_wav_mesh_bathytopo

File: .gitmodules [hunk 1]
```diff
@@ -10,7 +10,7 @@
 	url = git@github.com:BeTR-biogeochemistry-modeling/sbetr.git
 [submodule "components/mpas-source"]
 	path = components/mpas-source
-	url = git@github.com:MPAS-Dev/MPAS-Model.git
+	url = git@github.com:sbrus89/MPAS-Model.git
```
Comment (vanroekel) [inline .gitmodules:13]: is there a reason to point at your own fork here?

File: cime_config/config_grids.xml [hunk 2]
```diff
@@ -3935,12 +3934,22 @@
       <map name="ATM2WAV_SMAPNAME">cpl/gridmaps/T62/map_T62_TO_wQU100_bilin.20210202.nc</map>
     </gridmap>
 
-    <gridmap atm_grid="T62" wav_grid="wDE4">
-      <map name="ATM2WAV_SMAPNAME">cpl/gridmaps/T62/map_T62_TO_wDE4_blin.20200611.nc</map>
+    <gridmap atm_grid="T62" wav_grid="wQU225EC60to30">
```
Comment (vanroekel) [inline cime_config/config_grids.xml:3937]: are you planning to support jigsaw grids eventually?  I only see a v1 grid compset

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml [hunk 3]
```diff
@@ -229,6 +229,7 @@
 <config_cvmix_kpp_use_enhanced_diff>.true.</config_cvmix_kpp_use_enhanced_diff>
 <config_cvmix_kpp_nonlocal_with_implicit_mix>.false.</config_cvmix_kpp_nonlocal_with_implicit_mix>
 <config_cvmix_kpp_use_theory_wave>.false.</config_cvmix_kpp_use_theory_wave>
+<config_cvmix_kpp_use_active_wave>.false.</config_cvmix_kpp_use_active_wave>
```
Comment (vanroekel) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:232]: is this why we point to a separate fork as you don't have this merged to mpas yet?

**Review by vanroekel — COMMENTED** [review]
some questions mostly, but things look good overall.  Are you planning to do another PR once I add the new ocn component?

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml (hunk 3 above)
Comment (sbrus89) [reply components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:232]: @vanroekel, exactly. It's just a temporary until I complete the Stokes drift implementation and get it merged.

File: cime_config/config_grids.xml (hunk 2 above)
Comment (sbrus89) [reply cime_config/config_grids.xml:3937]: Yes definitely. I just did Ec60to30 to start.

File: .gitmodules (hunk 1 above)
Comment (sbrus89) [reply .gitmodules:13]: This is just temporary. See below.

**Comment by sbrus89** [comment]
Thanks for taking a look @vanroekel, Yes, I'll do another PR once the new component is added. I just wanted to sync this up with my development first.

**Review by nairita87 — APPROVED** [review]
Looks good to me, thanks!

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml (hunk 3 above)
Comment (vanroekel) [reply components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:232]: got it that's what I thought but thought I'd confirm.

**Review by vanroekel — APPROVED** [review]
Looks good to me, but would appreciate a final look from @jonbob before final integration.

File: .gitmodules [hunk 4]
```diff
@@ -10,7 +10,7 @@
...
 [submodule "externals/kokkos"]
 	path = externals/kokkos
 	url = git@github.com:E3SM-Project/kokkos.git
```
Comment (jonbob) [inline .gitmodules:16]: @sbrus89 - you'll have to rebase with the current CIME configuration, now that many parts have been brought into E3SM?

Comment (sbrus89) [reply .gitmodules:16]: @jonbob - yes, definitely. My plan was to do that in a separate PR following this one.

**Comment by sbrus89** [comment]
@jonbob, did you have any other feedback on this one?

**Comment by jonbob** [comment]
sorry -- no, otherwise it all looked good to me

File: .gitmodules (hunk 4 above)
Comment (jonbob) [reply .gitmodules:16]: sounds good

**Review by jonbob — APPROVED** [review]

**Comment by sbrus89** [comment]
thanks, @jonbob!

**Pull request merged**